client/src/context/UserContext.jsx

```diff
@@ -1,25 +1,32 @@
-import React, { createContext, useState, useContext } from 'react';
-
+import React, { createContext, useState, useContext, useEffect } from 'react';
 
 //create userContext for user data to be passed globally
 
 const UserContext = createContext();
 
 export const UserProvider = ({ children }) => {
   const [user, setUser] = useState(null); 
+
  
 // Set user data after successful login
   const login = (userData) => {
     setUser(userData);  
+    //localStorage.setItem('user', JSON.stringify(userData));
   };
 
   // Log out the current user
   const logout = () => {
+    const confirmed = window.confirm('Are you sure you want to log out?');
+    if (confirmed) {
+      // Clear user data and token from local storage
+    // localStorage.removeItem('user');
+    // localStorage.removeItem('token');
     setUser(null);
```

Comment:
It looks like you are using useContext and using setUser(null) to remove the user data/token; Lines 22-23 can be removed.